metricbeat/module/kubernetes/controllermanager/controllermanager_test.go

```diff
@@ -20,32 +20,35 @@
 package controllermanager
 
 import (
+	k "github.com/elastic/beats/v7/metricbeat/helper/kubernetes/ktest"
 	"testing"
 
 	"github.com/elastic/beats/v7/metricbeat/helper/prometheus/ptest"
 	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
 	_ "github.com/elastic/beats/v7/metricbeat/module/kubernetes"
 )
 
+var files = []string{
+	"./_meta/test/metrics.1.25",
+	"./_meta/test/metrics.1.26",
+	"./_meta/test/metrics.1.27",
+}
+
 func TestEventMapping(t *testing.T) {
-	ptest.TestMetricSet(t, "kubernetes", "controllermanager",
-		ptest.TestCases{
-			ptest.TestCase{
-				MetricsFile:  "./_meta/test/metrics.1.25",
-				ExpectedFile: "./_meta/test/metrics.1.25.expected",
-			},
-			ptest.TestCase{
-				MetricsFile:  "./_meta/test/metrics.1.26",
-				ExpectedFile: "./_meta/test/metrics.1.26.expected",
-			},
-			ptest.TestCase{
-				MetricsFile:  "./_meta/test/metrics.1.27",
-				ExpectedFile: "./_meta/test/metrics.1.27.expected",
-			},
-		},
-	)
+	var testCases ptest.TestCases
+	for _, file := range files {
+		testCases = append(testCases, ptest.TestCase{
+			MetricsFile:  file,
+			ExpectedFile: file + ".expected",
+		})
+	}
+	ptest.TestMetricSet(t, "kubernetes", "controllermanager", testCases)
 }
 
 func TestData(t *testing.T) {
 	mbtest.TestDataFiles(t, "kubernetes", "controllermanager")
 }
+
+func TestMetricsFamily(t *testing.T) {
+	k.TestStateMetricsFamily(t, files, mapping)
```

Comment:
That can be confusing since `controllermanager` and the rest control-plane are not retrieving metrics from kube-state-metrics. 
So I think that we should not use this helper function here. If `TestStateMetricsFamily` is generic and can be used for any kind of Prometheus metrics and not just for "state"metrics then I suggest we rename it properly and use it accordingly here too.

Reply:
You are right @ChrsMark , I didn't notice the name had `state` in it. I will change it, since the function is generic and update this PR

Reply:
I removed the `state` now in the name. Luckily the description of the function was very generic, so I updated only the name.